src/handler/http/request/alerts/destinations.rs

```diff
@@ -15,10 +15,24 @@
 
 use std::io::Error;
 
-use actix_web::{delete, get, http, post, put, web, HttpRequest, HttpResponse};
-use config::meta::alerts::destinations::Destination;
+use actix_web::{delete, get, post, put, web, HttpRequest, HttpResponse};
 
-use crate::{common::meta::http::HttpResponse as MetaHttpResponse, service::alerts::destinations};
+use crate::{
+    common::meta::http::HttpResponse as MetaHttpResponse,
+    handler::http::models::destinations::Destination,
+    service::{alerts::destinations, db::alerts::destinations::DestinationError},
+};
+
+impl From<DestinationError> for HttpResponse {
+    fn from(value: DestinationError) -> Self {
+        match value {
+            DestinationError::InUse(e) => MetaHttpResponse::conflict(e),
+            DestinationError::InfraError(e) => MetaHttpResponse::internal_error(e),
+            DestinationError::NotFound => MetaHttpResponse::not_found(DestinationError::NotFound),
+            other_err => MetaHttpResponse::bad_request(other_err),
```

Comment:
In general I'd probably expect that any sort of unexpected/other error would just be an internal server error rather than a bad request error. Typically I'd expect a bad request error only when validation of client input fails in some way.

Reply:
I went through all the variants of `DestinationError` and return specific `HttpResponse` for those special variants. The rest of the variants are all validation errors.